steps/index.lab.md

```diff
@@ -109,107 +108,140 @@ dependencies {
+### Connect your app
 
-If you get errors later in this codelab with the message "CONFIGURATION_NOT_FOUND", come back to this step and double check your work.
+In Android Studio, open `MainActivity.kt`, then add the following code inside the `onCreate` method:
 
-### Configure Realtime Database
+```kotlin
+// When running in debug mode, connect to the Firebase Emulator Suite.
+// "10.0.2.2" is a special IP address which allows the Android Emulator
+// to connect to "localhost" on the host computer. The port values (9xxx)
+// must match the values defined in the firebase.json file.
+if (BuildConfig.DEBUG) {
+    Firebase.database.useEmulator("10.0.2.2", 9000)
+    Firebase.auth.useEmulator("10.0.2.2", 9099)
+    Firebase.storage.useEmulator("10.0.2.2", 9199)
+}
+```
 
-As mentioned, this app will store chat messages in Firebase Realtime Database. In this step we will create a database and configure the security via a JSON configuration language called Security Rules.
+## Run the starter app
+Duration: 03:00
 
-1. Go to your project in the Firebase console and select **Realtime Database** from the left navigation.
-2. Click **Create Database** create a new Realtime Database instance and then select the `us-central1` region and click **Next**.
-3. When prompted about security rules, choose **locked mode** and click **Enable**. 
+### Add google-services.json
 
-Once the database has been created, select the **Rules** tab and update the rules configuration with the following:
+In order for your Android app to connect to Firebase, you must add a `google-services.json` file inside the `app` folder of your Android project. For the purposes of this codelab, we've provided a mock JSON file which will allow you to connect to the Firebase Emulator Suite.
 
+Copy the `mock-google-services.json` file into the `build-android-start/app` folder as `google-services.json`:
+
+```shell
+cp mock-google-services.json build-android-start/app/google-services.json
 ```
-{
-  "rules": {
-    "messages": {
-      ".read": "auth.uid != null",
-      ".write": "auth.uid != null"
-    }
-  }
-}
-```
 
-Click "Publish" to publish the new rules.
+In the final step of this codelab, you'll learn how to create a real Firebase project and Firebase Android App so that you can replace this mock JSON file with your own configuration.
+
+### Run the app
+
+Now that you've imported the project into Android Studio and added a Firebase configuration JSON file, you're ready to run the app for the first time.
 
-For more information on how this works (including documentation on the "auth" variable) see the Firebase  [security documentation](https://firebase.google.com/docs/database/security/quickstart).
+> aside negative
+>
+> Note: in order for your app to communicate with the Firebase Emulator Suite it must be running on an Android Emulator, not a real Android device. This will allow the app to communicate with the Firebase Emulator Suite on `localhost`.
```

Comment:
```suggestion
> Note: In order for your app to communicate with the Firebase Emulator Suite, it must be running on an Android Emulator, not a real Android device. This will allow the app to communicate with the Firebase Emulator Suite on `localhost`.
```